pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -916,11 +930,15 @@ void updateIdealStateOnSegmentCompletion(String realtimeTableName, String commit
             "Exceeded max segment completion time for segment " + committingSegmentName);
       }
       updateInstanceStatesForNewConsumingSegment(idealState.getRecord().getMapFields(), committingSegmentName,
-          newSegmentName, segmentAssignment, instancePartitionsMap);
+          isTablePaused(idealState) ? null : newSegmentName, segmentAssignment, instancePartitionsMap);
```

Comment:
This method is called with `newSegmentName` set to null `null` if the table is being paused right? Why check idealstate again?

Reply:
In regular cases, you're right. The new segment name in step 2 of the commit process will be null and here in step 3 we'll have a  null newSegmentName. In an edge case where pause is issued after step 2 and right before step 3, the new segment name is not null, but we need to pass in null to prevent creating new consuming segment in ideal state.